src/content/reference/react/use.md

```diff
@@ -0,0 +1,769 @@
+
+export function Message({ messagePromise }) {
+  const content = use(messagePromise);
+  return <p>Here is the message: {content}</p>;
+}
+```
+
+```js index.js hidden
+// TODO: update to import from stable
+// react instead of canary once the `use`
+// Hook is in a stable release of React
+import React, { StrictMode } from "react";
+import { createRoot } from "react-dom/client";
+import "./styles.css";
+
+// TODO: update this example to use
+// the Codesandbox Server Component
+// demo enviornemnt once it is created
+import App from "./App";
+
+const root = createRoot(document.getElementById("root"));
+root.render(
+  <StrictMode>
+    <App />
+  </StrictMode>
+);
+```
+
+```json package.json hidden
+{
+  "dependencies": {
+    "react": "18.3.0-canary-9377e1010-20230712",
+    "react-dom": "18.3.0-canary-9377e1010-20230712",
+    "react-scripts": "^5.0.0"
+  },
+  "main": "/index.js"
+}
+```
+</Sandpack>
+
+---
+
+## Troubleshooting {/*troubleshooting*/}
+
+### I'm getting this error: "An unsupported type was passed to use" {/*im-getting-this-error-an-unsupported-type-was-passed-to-use*/}
+
+The `use` hook only supports [JavaScript promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) and React's [Context](/learn/passing-data-deeply-with-context) as parameters.
+
+### I'm getting this error: "Suspense Exception: This is not a real error!" {/*suspense-exception-error*/}
+
+You are either calling `use` outside of a React component or Hook function, or calling `use` in a try-catch block. If you are calling `use` inside a `try-catch` block wrap your component in an error boundary, or call the promise's `catch` to catch the error and resolve the promise with another value. [See these examples](#dealing-with-rejected-promises).
+
+If you are calling `use` outside a React component or Hook function, move the `use` call to a React component or Hook function.
+
+```jsx
+function Message({messagePromise}) {
+  function download() {
+    // ❌ `use`'s parent function is not a Component or Hook!
+    const message = use(messagePromise);
+    //...
+```
+instead, call `use` outside any component closures, where the parent function is a component or Hook:
+```jsx
+function Message({messagePromise}) {
+  // ✅ This works! The parent function is a component. 
+  const message = use(messagePromise);
+  //...
+```
+
+
+
+### How do I prevent the component calling `use` from abruptly appearing in the UI? {/*how-do-i-prevent-the-component-calling-use-from-abruptly-appearing-in-the-ui*/}
+
+Components that pass a promise to `use` must be wrapped in [`Suspense`](/reference/react/Suspense) to prevent the Component from "popping in" when the promise is resolved. In the examples below, the `Message` Component calls the `use` hook. Because the `Message` Component is not wrapped in [`Suspense`](/reference/react/Suspense) the `Message` component will abruptly pop into the page when `messagePromise` is resolved:
```

Comment:
When we saying "popping in", are we talking about a layout jump?

Reply:
also confused about what this section means

Reply:
I've removed this section.